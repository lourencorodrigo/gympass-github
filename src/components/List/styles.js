import styled from "styled-components";

export const Ul = styled.ul`
  padding: 0;
`;

export const Li = styled.li;
