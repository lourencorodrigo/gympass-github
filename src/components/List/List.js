import React from "react";
import PropTypes from "prop-types";

import { Ul } from "./styles";
import ListItem from "./ListItem";

const List = ({ children }) => <Ul>{children}</Ul>;

List.propTypes = {
  children: PropTypes.any
};

List.Item = ListItem;

export default List;
