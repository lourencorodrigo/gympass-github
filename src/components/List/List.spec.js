import React from "react";
import ReactDOM from "react-dom";
import ReactTestRenderer from "react-test-renderer";
import { mount } from "enzyme";

import List from "./";

describe("List component", () => {
  const element = (
    <List>
      <List.Item>Test 1</List.Item>
      <List.Item>Test 2</List.Item>
    </List>
  );

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(element, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("should compare the text.", () => {
    const wrapper = mount(element);
    expect(wrapper.find("li").length).toEqual(2);
  });

  it("should compare the component with a snapshot", () => {
    const component = ReactTestRenderer.create(element);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
