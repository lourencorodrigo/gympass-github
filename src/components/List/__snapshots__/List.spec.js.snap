// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component should compare the component with a snapshot 1`] = `
<ul
  className="sc-bdVaJa lfmrhJ"
>
  <li
    className="sc-bwzfXH iWjNCq"
  >
    Test 1
  </li>
  <li
    className="sc-bwzfXH iWjNCq"
  >
    Test 2
  </li>
</ul>
`;
